ludwig/api.py

```diff
@@ -615,6 +631,19 @@ def on_epoch_end(self, trainer, progress_tracker, save_path):
                     if not path_exists(weights_save_path):
                         with open_file(weights_save_path, "wb") as f:
                             torch.save(self.model.state_dict(), f)
+
+                    # Save inference module
+                    if not skip_save_inference_module:
+                        logger.info("Attempting to save inference module with best weights from saved checkpoint...")
+                        inference_module_path = os.path.join(model_dir, INFERENCE_MODULE_FILE_NAME)
+                        try:
+                            self.save_torchscript(inference_module_path)
+                            logger.info(f'Saved inference module to: "{inference_module_path}"')
+                        except Exception:
+                            logger.warning(
```

Comment:
Swallowing the exception in all cases feels off to me, as the user may wish to explicitly save the model and fail loudly if it cannot.

An alternative we might consider here:

- `skip_save_inference_module` defaults to `None`, and if it's set to `None`, then it will attempt to save the inference module, but fail softly if it cannot.
- When set to `False`, fail loudly if the inference model cannot be saved. Also fail fast before training if set to `False` and `skip_save_model=True`.

What do you think?

Reply:
This sounds good to me. One thing I'd add is a case for if train is called with `skip_save_model=True` and `skip_save_inference_module=None`. 

In order to not break the API for current users/tests, I feel that in this case, `skip_save_inference_module` should be set to the same value as `skip_save_model` (i.e. True).